core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+        );
+      }
+    } else {
+      const publishButton = await this.page.$('.e2e-test-publish-exploration');
+      if (!publishButton) {
+        showMessage(
+          'Exploration is set to Public and is accessible to Oppia users.'
+        );
+      } else {
+        throw new Error(
+          'Exploration is set to Private and is not' +
+            ' accessible to Oppia users.'
+        );
+      }
+    }
+  }
+
+  /**
+   * This function helps in adding voice artist.
+   */
+  async addVoiceArtists(voiceArtists: string[]): Promise<void> {
+    for (let i = 0; i < voiceArtists.length; i++) {
+      await this.clickOn(voiceArtistEditButton);
+      await this.clickOn(addVoiceArtistUserName);
+      await this.page.waitForSelector(
+        `${addVoiceArtistUserName}:not([disabled])`
+      );
+      await this.page.click(addVoiceArtistUserName, {clickCount: 3});
+      await this.page.keyboard.press('Backspace');
+      await this.type(addVoiceArtistUserName, voiceArtists[i]);
+      await this.clickOn(voiceArtistSaveButton);
+
+      showMessage(voiceArtists[i] + ' has been added successfully.');
+    }
+  }
+
+  /**
+   * This function helps to choose notification type.
+   */
+  async optInToEmailNotifications(): Promise<void> {
+    await this.clickOn(feedbackToggleOff);
+  }
+
+  /**
+   * Exception function to verify the choice of receiving feedback
+   * and suggestion notifications via email
+   */
+  async expectEmailNotificationsToBeActivated(): Promise<void> {
+    await this.page.waitForSelector('input[id="feedback-switch"]');
+    const input = await this.page.$('input[id="feedback-switch"]');
+
+    if (!input) {
+      throw new Error('Feedback switch input element not found.');
+    }
+    const isChecked = await input.evaluate(
+      input => (input as HTMLInputElement).checked
+    );
+
+    if (!isChecked) {
+      showMessage('suggestions notifications via email are enabled.');
+    } else {
+      throw new Error('suggestions notifications via email are disabled.');
+    }
+  }
+
+  /**
+   * This funciton helps in deleting the exploration successfully.
+   */
+  async deleteExploration(): Promise<void> {
+    await this.page.waitForSelector(
+      `${deleteExplorationButton}:not([disabled]`
+    );
+    await this.clickOn(deleteExplorationButton);
+    await this.clickOn(deleteConfirmButton);
+  }
+
+  /**
+   * This function helps in verifying, if exploration is
+   * deleted successfully or not.
+   */
+  async expectExplorationToBeDeletedSuccessfullyFromCreatorDashboard(): Promise<void> {
+    try {
+      await this.goto(explorationUrlAfterPublished);
+      throw new Error('Exploration is not deleted successfully.');
+    } catch (error) {
+      showMessage('Exploration is successfully deleted.');
+    }
+  }
+
+  /**
+   * This function helps in drafting the exploration.
+   */
+  async saveDraftExploration(): Promise<void> {
+    await this.page.keyboard.press('Tab');
```

Comment:
After making any changes we need to do an extra click to save that change or to come out of the input box generally. So instead of an extra click we can use `Tab`.
For eg: if we are creating an exploration then we need to fill in all the details necessarily(title, Goal, Category, Language, Tags) after filling the last detail we need to do extra click to save that, Hence I used `Tab`.
This is valid with all.
Thanks.

Reply:
I see. I think readers of the code will have no idea why we're doing this though. Can you maybe have the mouse click on a neutral element on the page, or something like that, instead?

Also I feel like the methods for filling in details should be self-contained and "clean up after themselves", so if a save or extra click is needed to finalize it then do that at the end of the relevant method.